plugins/filter/filters.py

```diff
@@ -297,39 +316,41 @@ def client_properties(self, listener_dict, default_ssl_enabled, bouncy_castle_ke
             final_dict[config_prefix + 'ssl.keystore.type'] = 'BCFKS'
             final_dict[config_prefix + 'ssl.truststore.type'] = 'BCFKS'
 
-        if self.normalize_sasl_protocol(listener_dict.get('sasl_protocol', default_sasl_protocol)) == 'PLAIN' and not omit_jaas_configs:
+        sasl_protocols = listener_dict.get('sasl_protocol', default_sasl_protocol)
+        normalize_sasl_protocols = self.normalize_sasl_protocol(sasl_protocols)
+        if normalize_sasl_protocols[0] == 'PLAIN' and not omit_jaas_configs:
             final_dict[config_prefix + 'sasl.mechanism'] = 'PLAIN'
             final_dict[config_prefix + 'sasl.jaas.config'] = 'org.apache.kafka.common.security.plain.PlainLoginModule required username=\"' +\
                 sasl_plain_username +\
                 '\" password=\"' +\
                 str(sasl_plain_password) + '\";'
 
-        if self.normalize_sasl_protocol(listener_dict.get('sasl_protocol', default_sasl_protocol)) == 'SCRAM-SHA-512' and not omit_jaas_configs:
+        if normalize_sasl_protocols[0] == 'SCRAM-SHA-512' and not omit_jaas_configs and not kraft_listener:
```

Comment:
how do we handle the case when this kraft_listener is set to true ? Also when are we setting it to true?